fusillade/clouddirectory.py

```diff
@@ -1316,15 +1330,26 @@ def __init__(self, cloud_directory: CloudDirectory, name: str = None, object_ref
 
     def lookup_policies(self) -> List[str]:
         try:
-            if self.groups:
-                policy_paths = self._lookup_policies_threaded()
-            else:
-                policy_paths = self.cd.lookup_policy(self.object_ref)
+            policy_paths = self._lookup_policies_threaded()
         except cd_client.exceptions.ResourceNotFoundException:
             self.provision_user(self.cd, self.name)
-            policy_paths = self.cd.lookup_policy(self.object_ref)
+            policy_paths = self._lookup_policies_threaded()
         return self.cd.get_policies(policy_paths)
 
+    def lookup_policies_batched(self):
+        object_refs = self.groups + [self.object_ref]
+        operations = [self.cd.batch_lookup_policy(object_ref) for object_ref in object_refs]
```

Comment:
Can this list grow large enough to break the API call?

Reply:
This is never returned directly to the user. In backend, between fusillade and cloud directory this request can get too long. There is a batch_read request limit of 200 requests per batch request. When we start to approach that limit I will revisit.

Reply:
This would require group+role+policies > 200 for a single user which is not likely to happen. Resource policies and ownership will not be apart of the 200

Reply:
Consider adding a TODO with this information.